src/libcollections/vec_deque.rs

```diff
@@ -2401,4 +2438,24 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_from_vec() {
```

Comment:
Can you add some additional test cases here with varying vec lengths and capacities? It might be easiest to do something like:

```rust
#[test]
fn test_from_vec() {
    for cap in 0..35 {
        for len in 0..cap + 1 {
            let mut vec = Vec::with_capacity(cap);
            vec.extend(0..len);

            let vd = VecDeque::from(vec.clone());
            assert!(vd.cap().is_power_of_two());
            assert_eq!(vd.len(), vec.len());
            assert!(vd.into_iter().eq(vec));
        }
    }
}
```